src/App.tsx

```diff
@@ -1,14 +1,127 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext, useEffect, useMemo, useRef, useState,
+} from 'react';
+import { addTodo, deleteTodo, getTodos } from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+// eslint-disable-next-line max-len
+import { ErrorNotification } from './components/ErrorNotification/ErrorNotification';
+import { Footer } from './components/Footer/Footer';
+import { Header } from './components/Header/Header';
+import { TodoInfo } from './components/TodoInfo/TodoInfo';
+import { TodoList } from './components/TodoList/TodoList';
+import { FilterTypes } from './types/Enums';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filterType, setFilterType] = useState('All');
+  const [todoTitle, setTodoTitle] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const handleButtonClickAll = () => {
+    setFilterType(FilterTypes.All);
+  };
+
+  const handleButtonClickActive = () => {
+    setFilterType(FilterTypes.ACTIVE);
+  };
+
+  const handleButtonClickCompleted = () => {
+    setFilterType(FilterTypes.COMPLETED);
+  };
+
+  const onSubmit = useCallback((event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!todoTitle) {
+      setErrorMessage('Title can not be empty');
```

Comment:
It is good practice to move error text to the enum
```suggestion
      setErrorMessage(TodoErrors.EmptyTitle);
```